### PR TITLE
Migration to Codable

### PR DESCRIPTION
**Pull request by zdnk**
Resolves https://github.com/tuist/tuist/issues/59

### Short description 📝

Use Swifts native `Codable` instead of `JSONConvertible` abstraction.

### Solution 📦

Using implicit `CodingKeys` in most cases.
Solution in test is improvised and I am open to ideas how to change, redo or improve it.

### Implementation 👩‍💻👨‍💻

> Detail in a checklist the steps that you took to implement the PR.

- [x] Migrate models to `Codable`
- [x] Update tests
- [x] Update CHANGELOG

### CODE REVIEW

File: Tests/ProjectDescriptionTests/assertCodableEqualToJson.swift [hunk 1]
```diff
@@ -0,0 +1,18 @@
+//
```
Comment (pepibumur) [inline Tests/ProjectDescriptionTests/assertCodableEqualToJson.swift]: I'd remove these comments generated by Xcode

**Comment by zdnk** [comment]
I have updated he tests with fixes for the mentioned issue. However, there are still other tests failing which I do not know why yet. Maybe you can take a look.

**Comment by zdnk** [comment]
BTW, the SwiftShell dependency cannot be fetched. I have updated the revision to most recent one to make it work, so that might be the reason why are the other tests failing.

**Comment by pepibumur** [comment]
Hey @zdnk. I've merged the branch and fixed those tests on master. Basically, there was a property missing in the `JSONEnconder` that set it up to convert the keys to snake case. Everything is working fine now. Thank you very much for your contribution. I'll add you to a group of contributors so that you don't have to work from forks anymore.

**Comment by zdnk** [comment]
I see! Great, thanks :) Hopefully I will be able to contribute more!

**Pull request merged**